src/java.base/share/classes/sun/security/rsa/RSACore.java

```diff
@@ -403,55 +409,72 @@ BlindingRandomPair getBlindingRandomPair(
                 (this.d != null && this.d.equals(d))) {
 
                 BlindingRandomPair brp = null;
-                synchronized (this) {
-                    if (!u.equals(BigInteger.ZERO) &&
-                        !v.equals(BigInteger.ZERO)) {
-
-                        brp = new BlindingRandomPair(u, v);
-                        if (u.compareTo(BigInteger.ONE) <= 0 ||
-                            v.compareTo(BigInteger.ONE) <= 0) {
-
-                            // need to reset the random pair next time
-                            u = BigInteger.ZERO;
-                            v = BigInteger.ZERO;
-                        } else {
-                            u = u.modPow(BIG_TWO, n);
-                            v = v.modPow(BIG_TWO, n);
-                        }
-                    } // Otherwise, need to reset the random pair.
-                }
+                if (!u.equals(BigInteger.ZERO) &&
+                    !v.equals(BigInteger.ZERO)) {
+
```

Comment:
Is it ever possible that u could be equal to BigInteger.ZERO?  The only place I see a new BlindingRandomPair created is within the scope of the BlindingParameters object, and it has a guard in its constructor that resets u to BigInteger.ONE if u is zero.  Or is this check to guard against the reset of u/v down around line 416-419 (recalculated when reusable)?

Reply:
The both values are never zero now as the old code used zero to define the parameters needed to be reset.  I don't think this zero check is necessary.. let me try something else that fits with the isReusable() here and the one below when putting it into the queue